### PR TITLE
Setup MySQL backend

### PR DESCRIPTION
**Pull request by adetunjii**
Closes #3413.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [x] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by AlekSi — APPROVED** [review]

**Review by chilagrow — COMMENTED** [review]
Great work :hugs:! Please check the failing test.

**Comment by adetunjii** [comment]
> Great work 🤗! Please check the failing test.

@chilagrow I'm not sure what's causing that test to fail. I've not modified any code in that file, neither did I modify any code in relation to it.

File: internal/backends/mysql/query_iterator.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sync"
+
+	"github.com/FerretDB/FerretDB/internal/backends/postgresql/metadata"
```
Comment (AlekSi) [inline internal/backends/mysql/query_iterator.go:23]: That's the wrong import and the cause of the failed test

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/backends/mysql/query_iterator.go (hunk 1 above)
Comment (AlekSi) [inline internal/backends/mysql/query_iterator.go:23]:
```suggestion
	"github.com/FerretDB/FerretDB/internal/backends/mysql/metadata"
```

**Review by AlekSi — APPROVED** [review]
Fixing import fixed CI

**Pull request merged**